### PR TITLE
Add parameter to localise mobile menu toggle button

### LINKED ISSUE #1904: Make 'menu' text in header customisable
## What
Split out from https://github.com/alphagov/govuk-frontend/issues/1682

The 'menu' text in the header navigation toggle button currently cannot be customised using the header Nunjucks macro. We intended to fix this in https://github.com/alphagov/govuk-frontend/issues/1682, however doing so raises the issue of how should longer text in the menu button behave on smaller viewports (should the text wrap, move the button to a new line, how should the arrow in the button be aligned etc.)

I chatted with @christopherthomasdesign about this who thinks that there's potentially a couple of different scenarios to cover here which would create extra design and dev work. 

I'm thinking that it would be good to see some use cases of services needing to use different text here to understand the use case properly.

## Why
We might still want to do this but it would be good to triage it as a team.

## Who needs to know about this
Chris, devs

### PR DESCRIPTION
**Pull request by domoscargin**
Adds the `menuButtonText` parameter to customise the text of the menu toggle button that appears on narrow screens.

Cherry picked this from @querkmachine's spike [#2614] as the minimum changes required.

Fixes #1904

### CODE REVIEW

**Comment by 36degrees** [comment]
> ### HTML?
> We've allowed for text replacement. I can't immediately think of any good reason to allow for HTML here - there may be an aspect of the i18n work I'm missing though?

I'd suggest using `menuButtonText` so that if we need to introduce the HTML option in the future we can do so whilst being consistent with the other options and components.

(Also works if we find we need to add e.g. `menuButtonClasses` or `menuButtonAttributes`)

**Comment by querkmachine** [comment]
In my redo of the footer parameters I did end up changing them from taking strings to following the whole `html`/`text` separation, as it let me merge some related parameters (like copyright link, image toggles) into the options. 

I agree with @domoscargin that being able to set HTML here _probably_ isn't needed. The main counterpoint I can think of is a situation where a language may have quite a long word(s) for navigation menu, so they opt to abbreviate it but then want a title/aria-label/visually-hidden text to expand upon it.

**Comment by 36degrees** [comment]
Just noticed we now have a `navigationLabel` and a `menuButtonText` that In most cases I suspect should be the same, and both default to 'Menu'. Suspect the `navigationLabel` might easily be missed when translating as it's not visible text.

Wondering if it might make sense to make `navigationLabel` default to whatever `menuButtonText` is, so that they stay in sync unless someone intentionally sets `navigationLabel` to something different? 🤔

**Comment by domoscargin** [comment]
@36degrees, I've defaulted `navigationLabel` to `menuButtonText`.

I've also had a go at writing a changelog entry, but it feels a bit meh to just say "don't make it too long". Do we have any character length restrictions on other parameters?

**Review by 36degrees — COMMENTED** [review]
Changes all look sensible to me – my only suggestion is that it'd be good to have a test that the aria-label on the nav changes when `menuButtonText` is set.

**Comment by domoscargin** [comment]
I'm thinking it'd be enough documentation-wise to edit the header.yaml `menuButtonText` description to something like:

> Text for the button that toggles the navigation. This should be [the shortest useful text that describes the button's purpose][a maximum of 30 characters]. Defaults to 'Menu'.

**Comment by querkmachine** [comment]
To parrot what I've already said on Slack, for the permanent record:

> I’m not sure of what guidance should go with this. There is a limit, but the limit is a visual one—you can’t really describe it as ‘max n characters’, especially in a localisation context
> 
> It’s heavily dependent on the width of the device too. If we use 320px as a baseline then you have a decent amount of space, at least.

**Comment by domoscargin** [comment]
@christopherthomasdesign What do you think? Does this require more design work? Or will some kind of guidance do?

**Comment by christopherthomasdesign** [comment]
> @christopherthomasdesign What do you think? Does this require more design work? Or will some kind of guidance do?

I don't think so, long as we make it clear this is specifically to accommodate different languages. There seems to be enough space for a long-ish word to go alongside the logo. We just want to make sure users check the word they use still looks ok on smaller screens and at different zoom levels.

File: src/govuk/components/header/template.test.js [hunk 1]
```diff
@@ -136,6 +136,15 @@ describe('header', () => {
       expect($nav.attr('aria-label')).toEqual('Menu')
     })
 
+    it('renders navigation label correctly when custom menu button text is set', () => {
```
Comment (vanitabarrett) [inline src/govuk/components/header/template.test.js:139]: Do we have a test for when we're setting `menuButtonText` and `navigationLabel`?

Comment (domoscargin) [reply src/govuk/components/header/template.test.js:139]: We do now !

**Review by vanitabarrett — APPROVED** [review]
Looks good! Let's hold off merging until 4.3.1 is done

**Pull request merged**